torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1286,6 +1314,15 @@ RegisterOperators reg2({
           "aten::append( " decl_type "[](a!) self, " decl_type              \
           "(c) el) -> " decl_type "[](a!)",                                 \
           listAppend<Shared<c_type>, c_type::ElemType>),                    \
+      Operator(                                                             \
+          "aten::extend(" decl_type "[](a!) self, " decl_type               \
+          " [](b) other) -> ()",                                            \
+          listExtend<Shared<c_type>>),                                      \
+      Operator(                                                             \
+          "aten::copy(" decl_type                                           \
+          "[](a) self)"                                                     \
+          " -> " decl_type "[](b!)",                                        \
```

Comment:
The `!` annotation is only necessary to denote mutations to existing values. So you don't `(b!)` on the return type, since it's a new value returned by the op. The rules are this way so that pure ops (the common case) don't need annotations, e.g. (`add(Tensor a, Tensor b) -> Tensor`).

I know this is confusing! I am doing a writeup of all the aliasing/mutation semantics to help clarify

Reply:
Will fix

Reply:
Addressed